src/common/src/array/bytes_array.rs

```diff
@@ -36,8 +37,15 @@ impl Array for BytesArray {
     type Builder = BytesArrayBuilder;
     type Iter<'a> = ArrayIterator<'a, Self>;
     type OwnedItem = Box<[u8]>;
+    type RawIter<'a> = ArrayRawIter<'a, Self>;
     type RefItem<'a> = &'a [u8];
 
+    fn value_at_raw(&self, idx: usize) -> Self::RefItem<'_> {
+        let start = self.offset[idx];
+        let end = self.offset[idx + 1];
+        &self.data[start..end]
```

Comment:
I think we can trust the offsets and use `get_unchecked` here. BTW, this is similar to `non_null_value_unchecked`. How about merging them into one?